flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/table/CorrelateTest.scala

```diff
@@ -93,9 +98,25 @@ class CorrelateTest extends TableTestBase {
           "RecordType(INTEGER a, BIGINT b, VARCHAR(65536) c, VARCHAR(65536) s)"),
         term("joinType", "LEFT")
       ),
-      term("select", "c", "s")
+      term("select", "c", "s"),
+      term("where", ">(s, '')")
     )
 
     util.verifyTable(result, expected)
   }
+
+  /**
+    * Due to the improper translation of TableFunction left outer join (see CALCITE-2004), the
+    * join predicates can only be empty or literal true.
+    */
+  @Test (expected = classOf[ValidationException])
```

Comment:
There is a sub-package `validation` that contains tests that verify correct validation. Please move this test method into a new `CorrelateValidationTest` there.

Reply:
You can add a test for `leftOuterJoin(function(...) as 'x, true)` if you want to.